packages/terra-avatar/src/variants/Avatar.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import memoize from 'memoize-one';
+import 'terra-base/lib/baseStyles';
+import Utils from '../common/AvatarUtils';
+import styles from '../common/Avatar.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Specifies the alternative text for the image.
+   */
+  alt: PropTypes.string.isRequired,
+  /**
+   * Sets the background color. Defaults to `auto`. Accepted color variants are theme specific.
+   * One of: `'auto'`, `'neutral'`, `'one'`, `'two'`, `'three'`, `'four'`, `'five'`, `'six'`, `'seven'`, `'eight'`, `'nine'`, `'ten'`.
+   */
+  color: PropTypes.oneOf(['auto', 'neutral', 'one', 'two', 'three', 'four',
+    'five', 'six', 'seven', 'eight', 'nine', 'ten']),
+  /**
+   * Value used for the hash function when color is set to `auto`. If not provided, hash function utilizes alt.
+   */
+  hashValue: PropTypes.string,
+  /**
+   * The image to display.
+   */
+  image: PropTypes.string,
+  /**
+   * One or two letters to display.
+   */
+  initials: PropTypes.string,
+  /**
+   * Whether to hide avatar from the accessiblity tree.
+   */
+  isAriaHidden: PropTypes.bool,
+  /**
+   * Whether the person is deceased. Overrides any color variant.
+   */
+  isDeceased: PropTypes.bool,
+  /**
+   * Overrides the default size.
+   */
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  color: 'auto',
+  hashValue: undefined,
+  image: undefined,
+  initials: undefined,
+  isAriaHidden: false,
+  isDeceased: false,
+  size: undefined,
+};
+
+const ImageComponent = memoize(Utils.generateImage);
+
+const Avatar = ({
+  alt,
+  color,
+  hashValue,
+  image,
+  initials,
+  isAriaHidden,
+  isDeceased,
+  size,
+  ...customProps
+}) => {
+  let colorVariant;
+  let avatarContent;
+  if (image) {
+    colorVariant = '';
+    avatarContent = ImageComponent(image, alt, isAriaHidden, Utils.AVATAR_VARIANTS.USER);
+  } else if (initials && (initials.length === 1 || initials.length === 2)) {
+    colorVariant = Utils.setColor(alt, color, hashValue);
+    const avatarTextClassNames = cx('initials');
+    avatarContent = <span className={avatarTextClassNames} aria-hidden={isAriaHidden}>{initials.toUpperCase()}</span>;
+  } else {
+    colorVariant = Utils.setColor(alt, color, hashValue);
+    avatarContent = Utils.generateImagePlaceholder(alt, isAriaHidden, Utils.AVATAR_VARIANTS.USER);
+  }
+
+  const attributes = { ...customProps };
+  const avatarClassNames = cx([
+    'avatar',
+    `${colorVariant}`,
+    { image: Boolean(image) },
+    { 'is-deceased': isDeceased },
+    attributes.className,
+  ]);
+
+  return (
+    <div {...attributes} className={avatarClassNames} style={{ fontSize: size }}>
```

Comment:
I'd update this to something like this below to allow people to set additional inline styles if they need to:

```jsx
const customStyles = size ? Object.assign({ fontSize: size}, style) : style;

return (
  <div className={avatarClassNames} style={customStyles} {...attributes}>
    {avatarContent}
  </div>
);
```

Reply:
Ah, totally forgot. Do we expect inline styles to override the first class `size` prop? Or other way around? Based on the order of the `Object.assign` arguments, the inline style wins

Reply:
updated https://github.com/cerner/terra-core/pull/2061/commits/ba380de6fb977aeaee0a2ff74e5472c2c8413886

Reply:
I think how you have it in the updated code looks good to me.